src/eradius_log.erl

```diff
@@ -67,33 +68,35 @@ collect_meta({_NASIP, _NASPort, ReqID}, Request) ->
 collect_message({NASIP, NASPort, ReqID}, Request) ->
     io_lib:format("~s:~p [~p]: ~s",[inet:ntoa(NASIP), NASPort, ReqID, format_cmd(Request#radius_request.cmd)]).
 
+-spec reconfigure() -> ok.
+reconfigure() ->
+    gen_server:call(?SERVER, reconfigure).
+
 %%%===================================================================
 %%% gen_server callbacks
 %%%===================================================================
-init(_) ->
-    {ok, LogFile} = application:get_env(eradius, logfile),
-    filelib:ensure_dir(LogFile),
-    case file:open(LogFile, [append]) of
-    {ok, Fd} ->
-        {ok, Fd};
-    {error, eacces} ->
-        {stop, {file_permission_error, LogFile}};
-    Error ->
-        {stop, Error}
-    end.
+init(_) -> {ok, init_logger()}.
+
+handle_call(reconfigure, _From, State) ->
+    file:close(State),
+    {reply, ok, init_logger()};
+
+% for tests
+handle_call(get_state, _From, State) ->
+    {reply, State, State};
 
 handle_call(_Request, _From, State) ->
     {reply, ok, State}.
 
 handle_cast({write_request, Time, Sender, Request}, State) ->
     try
-	Msg = format_message(Time, Sender, Request),
-	ok = io:put_chars(State, Msg)
+        Msg = format_message(Time, Sender, Request),
+        ok = io:put_chars(State, Msg)
```

Comment:
You go though all the trouble (and work) for format a log message, when it might already be clear the the actuall put_chars will fail!?

There should be match that handle the case that state does not contain a valid open file.

BTW: the case that the log file is not open, might not be an error (but intended). Writing to lager:error in that case, will most certainly not the indented outcome.